### DISCUSSION TITLE
Water valve as switch

### DISCUSSION THREAD

**Discussion by sil3n7h** [iklein99/homebridge-smartthings#108] [Q&A]
Is there a way to convert the device type of a water valve to a switch? It appears that apple does not allow for automations on the valve type but does on the switch type? 

Thanks for any and all help!

My setup:
Homebridge on Mac mini
Aeotec SmartThings hub
Bulldog valve robot connects to hub >> homebridge >> HomeKit

**Comment by coryabooth** [comment]
Have you looking into Virtual Switches for Smartthings?

Perhaps this is a path to go vs. trying to convert your device.
You create a "fake" switch in Smartthings.
The switch alters its state based on criteria from the valve.

Homebridge uses the state of the "fake" switch to perform automations.

**Reply by sil3n7h** [reply]
Hey Cory, appreciate the reply. This is the exact route I ended up taking using this as a reference: https://www.youtube.com/watch?v=f_GHhipblNk&ab_channel=Bud%27sSmartHome

Everything appears to be working fine for now...
